src/libprojectM/Renderer/TextureManager.cpp

```diff
@@ -236,29 +228,38 @@ TextureSamplerDesc TextureManager::getTexture(const std::string fullName, const
 TextureSamplerDesc TextureManager::tryLoadingTexture(const std::string name)
 {
     TextureSamplerDesc texDesc;
-    GLint wrap_mode;
-    GLint filter_mode;
+    GLint wrapMode{0};
+    GLint filterMode{0};
     std::string unqualifiedName;
 
-    ExtractTextureSettings(name, wrap_mode, filter_mode, unqualifiedName);
+    ExtractTextureSettings(name, wrapMode, filterMode, unqualifiedName);
 
-    for (auto ext : extensions)
+    // Search order:
+    // 1. User preset dir
+    // 2. User texture dir
+    // 3. System/default preset dir
+    // 4. System/default texture dir
+    for (const auto& path : m_textureDirectories)
     {
-        std::string filename = unqualifiedName + ext;
-        std::string fullURL = presetsURL + pathSeparator + filename;
+        std::string fullPath = path;
+        fullPath += pathSeparator;
+        fullPath += unqualifiedName;
```

Comment:
Will introduce a clang-tidy warning for multiple memory reallocs, as each `+` operator creates a new string instance. Also, making it const would mean it would be impossible to add more dirs later, e.g. if the user adds a path in the frontend. There's no need to reinitialize projectM in this case.

Reply:
No reason to make it non-const preemptively. If that feature is needed in the future, just remove the const at that point.

Reply:
I think I've mixed up two things. Yes, having the `fullPath` variable const would sure be nice, but it's just a temporary loop var and not allocating, copying, allocating and copying again to add path segments is more important here IMO. Using the `+ ` operator will create & copy temporary string instances for each use, while the `+=` operator adds to the existing instance.